packages/build/src/core/normalize_flags.js

```diff
@@ -44,6 +44,7 @@ const getDefaultFlags = function ({ env: envOpt = {} }, combinedEnv) {
     functionsDistDir: DEFAULT_FUNCTIONS_DIST,
     cacheDir: DEFAULT_CACHE_DIR,
     deployId: combinedEnv.DEPLOY_ID,
+    buildId: combinedEnv.BUILD_ID,
```

Comment:
I realise `buildId` is not a valid flag so its "default value" will just be its value. But instead of accessing `env` directly I would much rather have all the `inputs`/config coming from a single point of truth, hence the new `buildId` prop here.

Reply:
Can we make it a flag? I think it might be nicer. For example, if we wanted to expose a build ID in local builds (which we currently don't at the moment), having a flag might be nice. Also, it makes the fact that we are using the build ID as input clearer (since environment variables are somewhat "global"/implicit).

I think you'd just need to add it to the list of CLI/yargs options.

Then we could add it to the buildbot like https://github.com/netlify/buildbot/blob/3aecc27b573adb3f8e700c38e918f994504826d8/script/run-build.sh#L79

Reply:
Also works for me 👍 I'm going to merge this PR and create a separate one for it if that's ok with you.

Reply:
Sounds good! :+1: